File: About-Swift-Advanced/Sources/Swift/ImportCAsOpaqueWrapper/ImportCAsOpaque.swift
import ImportCAsOpaque

public class ImportCAsOpaque {
  private let ref: ImportCAsOpaqueRef
  
  public init(value: Int32) {
    ref = ImportCAsOpaqueCreate(value)
  }
  
  public var value: Int32 {
    get {
      ImportCAsOpaqueGetValue(ref)
    }
    set {
      ImportCAsOpaqueSetValue(ref, newValue)
    }
  }
  
  deinit {
    ImportCAsOpaqueRelease(ref)
  }
}

File: About-Swift-Advanced/Sources/C/ImportCAsOpaque/ImportCAsOpaque.c
#include "ImportCAsOpaque.h"

struct ImportCAsOpaque {
  int32_t value;
};

ImportCAsOpaqueRef ImportCAsOpaqueCreate(int32_t value)
{
  ImportCAsOpaqueRef instance = calloc(1, sizeof(struct ImportCAsOpaque));
  instance->value = value;
  return instance;
}

int32_t ImportCAsOpaqueGetValue(ImportCAsOpaqueRef instance)
{
  return instance->value;
}

void ImportCAsOpaqueSetValue(ImportCAsOpaqueRef instance, int value)
{
  instance->value = value;
}

void ImportCAsOpaqueRelease(ImportCAsOpaqueRef instance)
{
  free(instance);
}
